[Tonishka Singh] iP

tonishka opened the pull request:
### YodaBot 👾 👾 

> "Do or do not. There is no try." - Yoda ([More about Yoda](https://www.starwars.com/databank/yoda))

If Star Wars fan you are, use this bot you must.  It is: 
- text-based
- easy to learn
- ~~fast~~ SUPER FAST to use
- in the **wise voice** of Yoda

YodaBot can: 
- [X] Manage and store your tasks
- [X] Categorise tasks based on type and status
- [X] Search for target tasks
- [ ] Send reminders (Soon 😃) 

`java -jar Yoda.jar`

```java
public static void main() {
    System.out.println("Wishes you good day, Yoda does.");
}
```

Commit:
Add Gradle support

tonishka committed:
Added user greeting

tonishka committed:
Added echo and exit

tonishka committed:
Added greet, echo, exit

tonishka committed:
Implemented skeletal version of Yoda

tonishka committed:
Text UI Enhancements

tonishka committed:
Added quests

tonishka committed:
Added Jedis

tonishka committed:
Added typewriter effect

tonishka committed:
Added feature to mark tasks done

tonishka committed:
Added feature to mark/unmark tasks

tonishka committed:
Added support for TODOs

tonishka committed:
Added support for events

tonishka committed:
Added support for deadlines

tonishka committed:
Handled incorrect user input exceptions

tonishka committed:
Handled more exceptions, added feature to delete tasks.

tonishka committed:
Enhanced Text UI Testing

tonishka committed:
Save data in local storage

tonishka committed:
Save data in local storage

tonishka committed:
Add support for date/time

tonishka committed:
Merge branch 'branch-Level-7'
Save user data in local storage

tonishka committed:
Resolve merge conflict with branch-Level-8

tonishka committed:
Add UI Class

tonishka committed:
Add more OOP

tonishka committed:
Add package

tonishka committed:
Minor bug fixes

tonishka committed:
Enhance save functionality

tonishka committed:
Find quests

tonishka committed:
Follow Java Coding Standard

tonishka committed:
Follow Coding Standard

tonishka committed:
Add JavaDoc comments

tonishka committed:
Merge branch 'branch-A-JavaDoc'
Merging with branch to add JavaDoc comments.

tonishka committed:
Resolve merge conflicts

tonishka committed:
Resolve merge conflicts

tonishka committed:
Add JUnit Tests

muraddurrani commented on src/main/java/yoda/QuestList.java:
```diff
@@ -0,0 +1,62 @@
+import java.util.ArrayList;
+
+public class QuestList {
+    private ArrayList<Quest> questList;
```
Should the name of this list be quests? I think the coding standards suggests that lists containing several items be named as a plural object.

muraddurrani commented on src/main/java/yoda/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+ * Represents a deadline bound quest.
+ * @author Tonishka Singh
+ */
+
```
I think there should not be a blank line between a JavaDoc comment and the method/class that it precedes. I noticed this in a few other places too.

muraddurrani commented on src/main/java/yoda/Parser.java:
```diff
@@ -0,0 +1,115 @@
+    public Parser() {}
+
+    /**
+     * Tokenizes and cleans the input by removing extraneous spaces.
+     * @param input Input string from user to clean.
+     * @return Array of cleaned String arguments.
```
I think there should be an empty line between the description and the parameters section. I noticed this in some other comments too.

muraddurrani commented on src/main/java/yoda/Parser.java:
```diff
@@ -0,0 +1,115 @@
+ */
+
+public class Parser {
+    public Parser() {}
```
Methods are supposed to be written with Egyptian brackets. Since this constructor is essentially default constructor, I suggest that you just delete it to avoid the trouble.

muraddurrani commented on src/main/java/yoda/Quest.java:
```diff
@@ -0,0 +1,93 @@
+        this.status = 1;
+    }
+
+    public void incompleteQuest() {
```
I think method names should be phrased as verbs. Perhaps completeQuest and incompleteQuest could be changed to something like markasComplete and markasIncomplete?

muraddurrani commented on src/main/java/yoda/Ui.java:
```diff
@@ -0,0 +1,78 @@
+ * @author Tonishka Singh
+ */
+public class Ui {
+    public Ui() {}
```
This is similar to the case with the constructor for Parser. I think you can just delete it to simplify things and avoid violations.

muraddurrani commented on src/main/java/yoda/Yoda.java:
```diff
@@ -0,0 +1,131 @@
+        System.out.print(questList.toString());
+    }
+
+    private void todo(String desc) {
```
Perhaps this could be changed to createTodo or makeTodo in order to make sure its phrased as a verb. The same applies for the following methods.

muraddurrani commented on src/main/java/yoda/QuestList.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    public ArrayList<Quest> find(String keyword) {
+        ArrayList<Quest> result = new ArrayList<>();
```
Perhaps the name could be changed to something like results or matches? List objects should be written as plural.

muraddurrani commented on src/main/java/yoda/Storage.java:
```diff
@@ -0,0 +1,112 @@
+
+    public ArrayList<Quest> load() {
+        Path path = Paths.get(dir, "YodaData");
+        boolean directoryExists = Files.exists(path);
```
The coding standard suggests that booleans be named to sound like booleans, so perhaps something like doesDirectoryExist?

muraddurrani commented on src/main/java/yoda/Ui.java:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * Prints the welcome message when user launches the application.
+     */
+    public void welcome() {
```
Maybe call this method printWelcome or displayWelcome or something along those lines to phrase it as a verb?

muraddurrani reviewed (commented):
I love how you have committed to the Yoda angle and how you've changed tasks to quests! Adds a lot of personality to the interaction.
Generally, I found the code fairly intuitive to read. I think the main things to focus on are slightly altering the JavaDocs and renaming some of the methods and field names to follow standards.

tonishka committed:
Minor bug fixes

tonishka committed:
Fixed formatting issues

tonishka committed:
Merge branch 'add-gradle-support' of https://github.com/tonishka/ip into branch-A-Gradle

tonishka committed:
Add Gradle Support

tonishka committed:
Merge branch 'branch-A-Gradle'

kavxya commented on src/main/java/Yoda.java:
```diff
@@ -1,26 +1,37 @@
 import java.util.Scanner;
 
 public class Yoda {
+    private enum YodaCommands {
```
I like how you made use of the enums for your bot.

kavxya commented on src/main/java/Yoda.java:
```diff
@@ -2,7 +2,7 @@
 public class Yoda {
     public static void main(String[] args) {
-
+        Jedi jedi = new Jedi();
```
May I know what the Jedi object is for?

kavxya commented on src/main/java/yoda/Ui.java:
```diff
@@ -43,6 +48,9 @@ private static void type(String text) {
+    /**
+     * Prints out the help message for user.
+     */
     public void help() {
```
Would the user benefit from receiving more statements in the help command?

tonishka committed:
Restructre code

tonishka committed:
Add Update functionality

tonishka committed:
Add Support for JavaFX

tonishka committed:
Level-10

tonishka committed:
Merge branch 'branch-Level-10'

tonishka committed:
Level 10

tonishka committed:
C-Update Extension

tonishka committed:
Merge branch 'branch-BCD-Extension'

tonishka committed:
C-Update Extension

tonishka committed:
Improve Code Quality

tonishka committed:
Add files via upload

tonishka committed:
Delete Ui.png

tonishka committed:
Use Assertions

tonishka committed:
Improve Code Quality

tonishka committed:
Merge pull request #2 from tonishka/branch-A-Assertions

Use Assertions

tonishka committed:
Update build.gradle

tonishka committed:
Merge branch 'branch-A-CodeQuality'

tonishka committed:
Merge pull request #3 from tonishka/branch-A-CodeQuality

Improve code quality

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Add Ui.png

tonishka committed:
Set theme jekyll-theme-minimal

tonishka committed:
Update README.md

tonishka committed:
Set theme jekyll-theme-midnight

tonishka committed:
Delete _config.yml

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Update README.md

tonishka committed:
Merge branch 'master' of https://github.com/tonishka/ip

tonishka committed:
Add User Guide

tonishka committed:
Update help command

Help command: Displays a long unformatted list of commands.

Product website has more detailed, better-formatted User Guide.
Long messages are tedious to read and make for sub-par UX.

Help command now redirects users to the product website.

In case of failure to redirect user, it provides the user
with link to manually access the help guide.

tonishka committed:
Refactor code
